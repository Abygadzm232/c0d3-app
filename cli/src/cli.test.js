import { exec } from 'child_process'
import { createCommand } from 'commander'

jest.mock('commander')

const program = {
  version: () => program,
  helpOption: () => program,
  usage: () => program,
  command: () => program,
  alias: () => program,
  description: () => program,
  option: () => program,
  action: () => program,
  parse: () => program,
  help: jest.fn(),
  args: []
}

createCommand.mockReturnValue(program)
const { init } = require('./cli')

describe('CLI c0d3', () => {
  test('Should display help', async () => {
    let { stdout } = await cli('c0d3')
    expect(stdout).toMatchSnapshot()
  })

  test('Should run c0d3', () => {
    expect(createCommand).toHaveBeenCalled()
  })

  test('Should run c0d3 --help', () => {
    init()
    expect(program.help).toHaveBeenCalled()
  })

  test('Should not run c0d3 --help', () => {
    program.args = ['submit']
    program.help = jest.fn()
    init()
    expect(program.help).not.toHaveBeenCalled()
  })
})

const cli = (args = '', cwd = '.') => {
  return new Promise(resolve => {
    exec(args, { cwd }, (error, stdout, stderr) => {
      resolve({
        code: error && error.code ? error.code : 0,
        error,
        stdout,
        stderr
      })
    })
  })
}
