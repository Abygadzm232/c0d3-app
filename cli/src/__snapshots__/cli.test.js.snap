// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI c0d3 Should display help 1`] = `
"Usage: c0d3 [options]

  [1m[35mA command line interface (CLI) for c0d3.com[39m[22m

Options:
  -V, --version       Show the c0d3 cli version
  -h, --help          Display help menu

Commands:
  submit|s [options]  Submits git diff of challenge to c0d3.com
  help [command]      display help for command
"
`;
